src/platformAccessory.ts

```diff
@@ -276,6 +279,13 @@ export class Zigbee2mqttAccessory implements BasicAccessory {
     }
   }
 
+  //Get QoS level, if force_disable_retain is set QoS is set to 0 level (best effort)
+  private getQosLevel(defaultQoS: QoS): IClientPublishOptions {
```

Comment:
I think it would be better to only return a `number` from this function  and use that in the other functions to set the `qos` option.
That way we can still change other options in the future, without touching this function.